src/eei.h

```diff
@@ -47,6 +50,42 @@ class WasmEngine {
   ) = 0;
 
   virtual void verifyContract(std::vector<uint8_t> const& code) = 0;
+
+  static void enableBenchmarking() noexcept { benchmarkingEnabled = true; }
+
+  void executionStarted() noexcept
+  {
+    if (benchmarkingEnabled)
+      executionStartTime = clock::now();
+  }
+
+  void executionFinished() noexcept
+  {
+    if (benchmarkingEnabled)
+    {
+      constexpr auto to_us = [](clock::duration d) {
+        return std::chrono::duration_cast<std::chrono::microseconds>(d).count();
+      };
+
+      auto executionDuration = clock::now() - executionStartTime;
```

Comment:
`executionDuration` is not interesting to me other than to compute invocation time as `executionDuration - instantiationDuration`. But `instantiationDuration` is interesting.